### ISSUE TITLE
Server docs jenkins jobs that builds PDF generate 429 

### ISSUE THREAD

**Issue by tbille** [canonical-web-and-design/ubuntu.com#8686]
It seems that we are getting a few 429 when building documentation for server docs on our jenkins instance. 

Maybe we should set to `sleep`  between requests to avoid facing this issue.

See: https://jenkins.canonical.com/webteam/job/upload-server-docs-pdf/configure

**Comment by darkalia** [comment]
Hello,

This is a consequence of another issue, which is more of a "design" one.
Some pages visited upon hitting www.ubuntu.com try to hit discourse.ubuntu.com to download some json file.

A fix has been committed yesterday to avoid hitting the rate limit, but the issue is that if discourse.ubuntu.com goes down, several pages are going to reply a 500 to the end user.
Content-cache won't help here, as it will try to validate the cache, sees a 500, because flask will have gotten a 500 from discourse and the end user will end up with an error.

I think we need a sort of cache for every request made to discourse.ubuntu.com and fall back on the last know working copy on the flask side.

What do you think ?

**Comment by tbille** [comment]
I removed the auth calls that we were making on discourse all discourse topics are public.
The auth calls were generating 429.

@darkalia since this is not related I opened another issue for server side caching: https://github.com/canonical-web-and-design/ubuntu.com/issues/8687

**Comment by nottrobin** [comment]
> Maybe we should set to sleep between requests to avoid facing this issue.

I think we can do a little better than that. I've written python code before to parse the 429 responses from discourse and wait for exactly the specified time before trying again, which I think is in [discli](https://github.com/howbazaar/discli).

> Content-cache won't help here, as it will try to validate the cache, sees a 500

I believe the content cache can help through careful use of the stale-if-error cache control instruction (I'm 99% sure content cache supports it), which we can look into.

I also believe this is the right way to solve it. We've tried caching at the flask later before and it was a mess. Aside from the general problem of multiple caches compounding invalidation issues which always leads to a bunch of problems, there's the issue of there being many replicas of the Flask app, which also compounds things, memory issues etc etc.

Basically, since we have a cache later already, we should leverage it as much as we possibly can rather than adding more, and I believe in this case we absolutely can.
